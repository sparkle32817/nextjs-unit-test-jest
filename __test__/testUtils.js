import { Provider } from "react-redux";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import { render } from "@testing-library/react";
import user from "@testing-library/user-event";

import { store } from "src/store";

export const push = jest.fn().mockResolvedValue(true);
const routerMock = {
  basePath: "",
  pathname: "/",
  route: "/",
  asPath: "/",
  query: {},
  locale: "",
  locales: [],
  defaultLocale: "",
  domainLocales: [],
  isLocaleDomain: false,
  push,
  replace: jest.fn().mockResolvedValue(true),
  reload: jest.fn(),
  back: jest.fn(),
  prefetch: jest.fn().mockResolvedValue(undefined),
  beforePopState: jest.fn(),
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  isFallback: false,
};

const customRender = (ui, value = routerMock) => {
  return render(
    <Provider store={store}>
      <RouterContext.Provider value={value}>{ui}</RouterContext.Provider>
    </Provider>
  );
};
export * from "@testing-library/react";
export * from 'react-dom/test-utils';

export { customRender as render, user };
