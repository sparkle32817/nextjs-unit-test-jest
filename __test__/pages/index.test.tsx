import React from "react";
import { render } from "../testUtils";
import Index from "pages/index";

describe("Loading component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Index />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
