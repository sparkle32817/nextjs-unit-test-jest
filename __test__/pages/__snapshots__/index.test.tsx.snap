// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="testing-production"
    >
      This is test page
    </div>
    <div
      style="width: 500px; height: 500px; display: flex; align-items: center; justify-content: center;"
    >
      <span
        class="css-1683kmo"
      >
        <span
          class="css-1rjycqt"
        />
        <span
          class="css-181511v"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
